packages/pipeline/README.md

```diff
@@ -46,4 +46,82 @@ npx projen new --from aws-prototyping-sdk pdk-pipeline-java
 
 ### CDK Nag
 
-In order to keep CDK Nag happy, make sure you build the pipeline before synth as per https://github.com/aws/aws-cdk/issues/18440.
\ No newline at end of file
+In order to keep CDK Nag happy, make sure you build the pipeline before synth as per https://github.com/aws/aws-cdk/issues/18440.
+
+## Multi-branch pipeline management
+
+If your team follows [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow), the pipelines module can optionally help you create independent environments to test and validate changes before merging. When you create a new branch, it will automatically create a new pipeline stack and any stages you configure. Once you have finished testing and delete the branch, it will automatically clean up the stacks created in the branch's environment.
+
+The feature is enabled and configured by setting the `branchNamePrefixes` property of the `PDKPipeline` construct. Any branches created matching this list of prefixes will create a new pipeline and stack.
+
+When your PDKPipeline is run, the current branch will be available in the `BRANCH` environment variable. You can use this to give unique names to the stacks and stages created by that branch. You can also enable and disable stages based on the branch name. For example, you may want the PipelineStack and Dev stage to get created for any branch and only create the Prod stage in the default branch.
+
+### PDKPipeline configuration
+
+#### Example: All Branches
+
+pipeline-stack.ts
+
+```ts
+this.pipeline = new PDKPipeline(this, "ApplicationPipeline", {
+  primarySynthDirectory: "packages/backend/cdk.out",
+  repositoryName: this.node.tryGetContext("repositoryName") || "monorepo",
+  branchNamePrefixes: [""],
```

Comment:
It might be a little more readable if we exposed a constant like `PDKPipeline.ALL_BRANCHES`? `[""]` makes logical sense but is a little harder to reason about at first glance :)